rosidl_generator_py/resource/_srv.py.em

```diff
@@ -1,7 +1,43 @@
 # generated from rosidl_generator_py/resource/_srv.py.em
 # generated code does not contain a copyright notice
 
-from @(package_name).msg._@(spec.srv_name)__request import @(spec.srv_name)_Request
-from @(package_name).msg._@(spec.srv_name)__response import @(spec.srv_name)_Response
-@(spec.srv_name)_Request  # unused
-@(spec.srv_name)_Response  # unused
+import logging
+import traceback
+
+
+class Metaclass(type):
+    """Metaclass of message '@(spec.srv_name)'."""
+    _TYPE_SUPPORT = None
+
+    @@classmethod
+    def __import_type_support__(cls):
+        __type_support_importable = False
+        try:
+            import rclpy
+            from rosidl_generator_py import import_type_support
+            __type_support_importable = True
+        except ImportError:
+            logger = logging.getLogger('rosidl_generator_py.@(spec.srv_name)')
+            logger.debug(
+                'Failed to import needed modules for type support:\n' + traceback.format_exc())
+
+        if __type_support_importable:
```

Comment:
No, putting the code following this line in an `else` block. The `try` block should always be as small as possible in order to not catch exceptions which shouldn't be catched.